### ISSUE TITLE
FR: Video preview with dual monitor setup

### ISSUE THREAD

**Issue by noembryo** [SubtitleEdit/subtitleedit#8358]
Not an issue, just a feature request.
Can we have the video preview in a different monitor than the application's main window?
This way, it would be easy to actually watch the video while fixing the subtitles and at the same time free up some precious space for the rest of the app's controls.
 Just a thought... 😉

**Comment by GrampaWildWilly** [comment]
This is already possible.  I've been doing it for years.  Just undock the video controls, then move the video player to the other monitor.

**Comment by noembryo** [comment]
I had tried this, but there are some problems that makes less than ideal.

First of all, it un-docks the audio panel too (and another small panel with a checkbox!?)
The extra panel must go somewhere and I haven't found a way to put them somewhere that makes me happy.
Also I was thinking of a full screen video player that I can toggle on and off when I want to use a dual monitor setup or a single monitor setup.
Another problem is the font size.
I have to change it every time I change configuration.
Too much work and the result is not that great.

Ideally I would like a full screen Subtitle Edit app on one monitor and a full screen player on the other.
And the ability to toggle that, with the normal app setup I use most of the time (the app window full screen on one monitor).
..but one can dream.. 😄 

The app is great anyway, this is just some extra icing..

**Comment by GrampaWildWilly** [comment]
![#01](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/56150289-eaa5-45a3-879a-fdce4069fb14)
![#02](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/88bc2345-15b4-4775-a3b2-bd18d36be610)
![#03](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/84a42fb1-9bf8-4b9f-8fcd-82b71418fcb1)
![#04](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/399fc16d-702c-47eb-9b50-5b0a461fdda7)

Now, this next bit is not straightforward.  I would very much like Nik (or somebody) to look into this.  I had to fiddle & fiddle & fiddle with this.  I maximized the video window, restored it, moved it from one monitor to the other & back again, closed & reopened SE several times.  Eventually, I hit Alt+Enter in the main SE window.  That is the default keyboard shortcut that puts the video player into fullscreen.  There were several attempts to do this that left the standard Windows title bar visible at the top of the screen & the video player controls visible at the bottom.  They did not disappear after a few seconds of playback.  But eventually, Alt+Enter put the MPV window into fullscreen on my second monitor & it looked like this:

![#05](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/6579d95d-9ba1-469a-9104-81feb40b8f6f)

In case you're wondering, this is an opera with English subtitles.

But once I close SE, the next time I open it, I'm going to have to fight with it to make the video player go fullscreen.  I wouldn't call this a bug, exactly.  But it is extremely annoying.  If it's at all relevant, I am running Windows 11 Pro 64-bits.

Now, I don't think you can toggle between single-monitor mode & dual-monitor mode & have the video player do what you want.  You can probably (I've never tried it) swap the fullscreen video player from one monitor to the other.  But if you have it on your primary monitor, it will of course cover the other SE windows, so I don't see the use in that.  All I know that you can do is to dock & undock the video controls.

I can't address your issue with fonts.  I'm not having any font issues, but then I'm just another user.  You'll have to rely on somebody with more inside knowledge than I to address that.  I would recommend, however, that you post a series of screenshots illustrating exactly what you're talking about.  To me, as a layman, your problem description dealing with fonts seems terribly insufficient for someone to take any action on.

**Comment by noembryo** [comment]
Thank you for your examples.
This is more or less my findings too (with the exception of the Alt+Enter tip which I never would have guessed).
The problem with the fonts is that there are very small (compared to the size of the full screen frame), but if I enlarge them, when I re-dock the panels they are huge (compared to the much smaller frame of the video panel inside the app)
I will try the mpv player to see if its better than the native direct show that I'm using now..

If this type of usage got a little more love, it may be workable, who knows..

**Comment by GrampaWildWilly** [comment]
My practice is to never redock the video controls.   But like I say, I'm just another user.  I think the original conception of the product was that you would use it either docked or undocked & not swap between the two.  All I know is I never redock the controls.  I want them undocked.  Always.

**Comment by noembryo** [comment]
Downloaded the mpv player, and it solved the font size problem (it resize them according to frame size).
Also the video quality is much better.
The only missing thing is a way to toggle the layout from dual screen to single screen.
I have to re-arrange everything manually every time I switch.
But, until further notice I can work with this..
Thank you for the hint.. 👍

**Comment by niksedk** [comment]
Not too intuitive, but you can also try the "full screen video player" function...

**Comment by noembryo** [comment]
...and how one can do this?
I looked at the menus/Settings and couldn't locate it..
Must be well hidden I suppose.. 😉

**Comment by niksedk** [comment]
He, yes.

It's here: 
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/a8e8aefa-d772-409a-a905-eb9598cc1927)


And try with SE on your main screen... and on your secondary screen...  maybe something could be made to make this easier.

**Comment by noembryo** [comment]
Hey, this is great! 🥳 
This is exactly what I wanted!

OK. And now you are here..
Is there a way for the full screen transport to _not_ hide the subtitles?
Or the subtitles to move a little up?
I used the "Vertical margin" in settings, but it doesn't seem to work..
I don't use the "mpv handles preview text" after all, because it makes the size very small when not in full screen..

Even like this though, its nice! 👍

**Comment by niksedk** [comment]
> OK. And now you are here.. Is there a way for the full screen transport to _not_ hide the subtitles? 

Where exactly? A screenshot?


>Or the subtitles to move a little up? I used the "Vertical margin" in settings...
Could be this one?
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/4f676f93-03b9-4261-90d5-5b1d4540e043)

**Comment by noembryo** [comment]
> Where exactly? A screenshot?

![err1](https://github.com/SubtitleEdit/subtitleedit/assets/24675403/437b5bb8-4b83-413d-935f-175df007a2e6)

There is another line of text under the one that is displayed.
> Could be this one?

I don't understand.
Where are those settings?
I used this one
![err2](https://github.com/SubtitleEdit/subtitleedit/assets/24675403/0d22b93c-23f1-4582-99aa-541bbc20e059)

**Comment by niksedk** [comment]
Try to set `mpv handles preview text` on

**Comment by noembryo** [comment]
Yeah, as I wrote,  I don't use the "mpv handles preview text" after all, because it makes the font size very small when not in full screen..
But even with it turned on I only get this

![err3](https://github.com/SubtitleEdit/subtitleedit/assets/24675403/940c3750-ed45-41a1-9610-632f70f40ac2)

Nothing like your screenshot style settings..
Is it from another app?
Anyway, with the "mpv handles preview text" turned on, the "Vertical margin" works, but apart from the size problem, it also puts the subs inside the video frame, a thing I try to avoid.
It would be nice if it had negative values to place the subs underneath the video frame, if there is space.

Also, If the "Vertical margin" setting doesn't work without the "mpv handles preview text", shouldn't get hidden when this setting is off?

**Comment by niksedk** [comment]
> Also, If the "Vertical margin" setting doesn't work without the "mpv handles preview text", shouldn't get hidden when this setting is off?

yes...

**Comment by GrampaWildWilly** [comment]
I find that Alt+Enter works only after much fiddling around.  With MPV.  And with MPV handling preview text.  Unlike the other poster, I'm perfectly happy to have the subtitles occupy space within the video aperture.  But my issue is that Alt+Enter just about never works on the first try for me.  This has been true for a while, but certainly true in the images I posted upthread here.  I keep getting this:

![#01](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/4d3cae09-31f5-4686-86f9-b5325d838238)

Rather pathologically, only the X is visible in the upper right corner of the window.  Standard Windows applications also show the Minimize & Maximize controls up there beside the X.  Like this:

![image](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/c9e98315-a31c-4a7e-86ae-ba9ffcb5d674)

The MPV player doesn't have them.  A minor issue that I actually don't care about.  But I think it's all part of the pathological behavior I'm reporting here.

I closed & reopened SE several times.  I maximized & restored the video player using the Windows controls several times.  Alt+Enter several times.  Things continued to look as shown in that first image in this post.

Here's what finally worked.  With the video player occupying an area in the middle of my second monitor, meaning not maximized, I manually dragged the left & right edges of the window to the edges of the monitor with my mouse.  Then I dragged the top edge of the window to the top edge of the monitor.  Windows automatically snaps the bottom edge of the window to the bottom edge of the monitor when you do that.  Then Alt+Enter in the main SE window made it look like this:

![#02](https://github.com/SubtitleEdit/subtitleedit/assets/75588964/9b55f140-03ab-4f8e-a9f0-1365bbb013f6)

This is what it's supposed to look like.  No Windows title bar at the top.  No video player controls at the bottom.  I can make the video player controls appear by hovering the mouse in the player window.  This is the way it's supposed to work.  When I move the mouse out of the video window, the player controls disappear.  Once again, this is how it's supposed to work.

But when I close & reopen SE, the problem is back.  The window title bar is visible at the top of the player & the player controls are once again visible at the bottom.  My procedure of starting with a non-maximized video player & dragging the edges did work again . . . on about the third or fourth attempt.

This should not be this hard.  The window should just go fullscreen on the first attempt at Alt+Enter.  The fullscreen state should be remembered across closing & reopening SE.  Fiddling should not be required.

**Comment by noembryo** [comment]
> But when I close & reopen SE, the problem is back.

Yes, this is what happens to me too.
If you close the program in full-screen mode, it doesn't stay in full screen mode.
Now that I think about it though, I don't know any program that does that.
If re-dock them before closing and un-dock them after next opening, all works as they should.

But, I don't see why you bother and not using the Full Screen button on the transport.
It's really a life-savior.. 😃

**Comment by GrampaWildWilly** [comment]
When I have the MPV player in proper fullscreen after all that fiddling, everything runs the way it's supposed to.  When I close SE & reopen it, the MPV player opens on the second monitor maximized.  Not in proper fullscreen mode, but maximized as recognized by Windows.  The system menu, the one you get from Alt+Spacebar has the menu entry Maximize grayed out, unavailable, because the window is already maximized.  Restore is available but Maximize is not.  So after close & reopen, MPV is indeed maximized.  But maximized is not fullscreen.  The difference is subtle.  Look at the 2 images in my preceding post.  When the window is maximized, the title bar & the player controls are visible.  When the window is in proper fullscreen, those controls are not visible.  I say it should be possible to have MPV in proper fullscreen, close SE, reopen SE, & MPV should still be in fullscreen.  Not maximized, as happens now, but fullscreen.  It was fullscreen before I close SE.  It should open in fullscreen when I reopen SE.  Seems to me that many releases back, that was how SE did work.  I'm asking for that functionality to be restored.  I'm also asking that it should not be necessary to fiddle with it get it into fullscreen.  Alt+Enter should just work, first time, every time.

**Comment by LeonCheung** [comment]
AFAIK some video editor supports, like CapCut:
![20240516011310](https://github.com/SubtitleEdit/subtitleedit/assets/494788/3dd5a07c-80d0-4c0f-bf72-536a63347930)

Say, if you select the one called "Player first" then it detects another valid monitor and displays the player as fullscreen there.

If it is possible, I mean "if" XD, then maybe it is only useful for layout 6 or 12 I guess?

![demo](https://github.com/SubtitleEdit/subtitleedit/assets/494788/6a98bc3d-04a7-4b78-900a-ec1e490d8498)

However, not sure if anyone also wants to have the list view or waveform being in fullscreen on a second monitor. XD

**Comment by GrampaWildWilly** [comment]
I have read the posts on here about the new ways to select your layout.  I know it's a relatively new way of doing things.  But I've never used one of them.  I'm still using the layout that has automatically migrated with me from . . . I don't know.  Whatever release of SE was around . . . 4 years ago?  5 years ago?  I don't seem to have any bread crumbs here to tell me when I started using SE.  But it was definitely at least SE 3 & maybe even SE 2.  I think I had the idea to use my second monitor within about 3 days of starting to use SE, so I've been running undocked for what may as well be forever.  I adjusted my layout as you can see in my post upthread here dated May 12.  It's my second post in this thread.  So what I am running now is based on a legacy layout I chose a long time ago.  I've never even opened the new layout picker.

I don't think the layout picker is really pertinent to the fact that Alt+Enter requires endless fiddling.  Nik told me way back when to use MPV so I always have.  I'm pretty sure the older releases didn't suffer this problem.

I also need to thank Nik again (totally off topic now) for telling me about Notepad++ a few years ago.  That could be the best advice I've ever had.  Or at least, I've rarely gotten better advice.

**Comment by GrampaWildWilly** [comment]
Further to my obtaining SE 4.0.6 beta 34 just now, the issue with full screen vs maximized still exists.  Closing & reopening SE does not remember that the MPV player window was in full screen mode on my second monitor.  It reopens with MPV in maximized state, not full screen.  However, it took me only a couple of cycles through Restore/Alt+Enter, without having to close & reopen SE several times, to correctly put the MPV player into proper full screen mode.  I suppose that's progress.  Still, I say there is more progress to be made.

**Comment by GrampaWildWilly** [comment]
I have had a couple of the recent betas & the situation seems to be greatly improved.  The process goes like this:

1. Open SE.
2. The MPV video player opens on the second monitor, where it was the preceding time I closed SE.
3. The video player is in Windows maximized state.  When I previously closed SE, the video player was in fullscreen.  I don't know what other terminology to use.  Maximized & fullscreen are NOT the same.
4. Open the Windows context menu available via Alt+Spacebar.
5. Execute the Windows Restore function on that menu to restore the video player window to no longer maximized.
6. Alt+Enter, which is SE's shortcut for putting the video player into fullscreen.

This has been working on the first try every time I open SE.  I'm up to 4.0.6 beta 131 now & it's quite reliable.  No more fiddling like I describe upthread here.  It is my recollection that past releases of SE, like maybe 2 or 3 years ago, would open with the video player already in fullscreen.  This little dodge did not used to be necessary.  But the current state of affairs is at least easy to fix & the behavior is now consistent, even if I still think it should open in fullscreen & not maximized.

I would not mind if this got closed, but I didn't open this so mine is not the only opinion that matters.
